lib/preprocess.js

```diff
@@ -16,16 +16,23 @@ export class Measurement {
     this.spark_version = m.spark_version
     this.fraudAssessment = null
     this.inet_group = m.inet_group
-    this.finished_at = m.finished_at
+    this.finished_at = parseDateTime(m.finished_at)
     this.provider_address = m.provider_address
     this.protocol = m.protocol
     this.byte_length = m.byte_length
-    this.start_at = m.start_at
-    this.first_byte_at = m.first_byte_at
-    this.end_at = m.end_at
+    this.start_at = parseDateTime(m.start_at)
+    this.first_byte_at = parseDateTime(m.first_byte_at)
+    this.end_at = parseDateTime(m.end_at)
   }
 }
 
+const parseDateTime = (str) => {
+  if (!str) return undefined
+  const value = new Date(str)
+  if (Number.isNaN(value.getTime())) return undefined
```

Comment:
Hmm, we should reject measurements with an invalid date. I think it's ok to throw an error here.